src/message_pool/msgpool/msg_pool.rs

```diff
@@ -49,30 +49,42 @@ use crate::message_pool::{
 const BLS_SIG_CACHE_SIZE: NonZeroUsize = nonzero!(40000usize);
 const SIG_VAL_CACHE_SIZE: NonZeroUsize = nonzero!(32000usize);
 
+pub const MAX_ACTOR_PENDING_MESSAGES: u64 = 1000;
+const MAX_UNTRUSTED_ACTOR_PENDING_MESSAGES: u64 = 10;
+
 /// Simple structure that contains a hash-map of messages where k: a message
 /// from address, v: a message which corresponds to that address.
 #[derive(Clone, Default, Debug)]
 pub struct MsgSet {
     pub(in crate::message_pool) msgs: HashMap<u64, SignedMessage>,
     next_sequence: u64,
+    max_actor_pending_messages: u64,
 }
 
 impl MsgSet {
     /// Generate a new `MsgSet` with an empty hash-map and setting the sequence
     /// specifically.
-    pub fn new(sequence: u64) -> Self {
+    pub fn new(sequence: u64, max_actor_pending_messages: u64) -> Self {
         MsgSet {
             msgs: HashMap::new(),
             next_sequence: sequence,
+            max_actor_pending_messages,
         }
     }
 
     /// Add a signed message to the `MsgSet`. Increase `next_sequence` if the
     /// message has a sequence greater than any existing message sequence.
-    pub fn add(&mut self, m: SignedMessage) -> Result<(), Error> {
+    pub fn add(&mut self, m: SignedMessage, untrusted: bool) -> Result<(), Error> {
```

Comment:
It's a bit hard to understand methods with boolean parameters. Without going to the definition of this method, I have no clue what is `true` in `.add(msg, true)` stands for. For ease of reading, it might be worth creating wrapper methods `add_trusted` and `add_untrusted` and exposing only them.

Reply:
I was hesitating using some `enum` actually.

Reply:
I removed some unnecessary field as well in `MsgSet`.